### PR TITLE
Improving EventDeduplicationLambdaIntegrationTest

### PR DESCRIPTION
**Pull request by mjsax**
- some more cleanup for KIP-182

### CODE REVIEW

File: src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java [hunk 1]
```diff
@@ -229,7 +232,10 @@ static KafkaStreams buildTopArticlesStream(final String bootstrapServers,
           return queue;
         },
 
-        new PriorityQueueSerde<>(comparator, valueAvroSerde)
+        // TODO: need to change this, after Damian's PR got merged to AK
```
Comment (miguno) [inline src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: What needs to be changed here?  I'd also reference the actual PR, not just "Damian's PR". :)

File: src/main/scala/io/confluent/examples/streams/algebird/CMSStore.scala [hunk 2]
```diff
@@ -146,6 +145,8 @@ class CMSStore[T: CMSHasher](override val name: String,
     */
   private var cms: TopCMS[T] = cmsMonoid.zero
 
+  private var lastUpdate: Long = 0L
```
Comment (miguno) [inline src/main/scala/io/confluent/examples/streams/algebird/CMSStore.scala]: For my understanding: Why do we need this now?

Comment (miguno) [reply src/main/scala/io/confluent/examples/streams/algebird/CMSStore.scala]: Ah, I see it's a timestamp.  I'd suggest to rename `lastUpdate` to sth more descriptive like `timestampOfLastStateStoreUpdate` (or better/shorter).

File: src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java [hunk 3]
```diff
@@ -220,10 +222,19 @@ public void shouldRemoveDuplicatesFromTheInput() throws Exception {
     // on de-duplicating late-arriving records.
     long maintainDurationPerEventInMs = TimeUnit.MINUTES.toMillis(10);
 
+    // The number of segments has no impact on "correctness".
+    // Using more segments implies larger overhead but allows for more fined grained record expiration
+    // Note: the specified retention time, is a _minimum_ time span and no strict upper time bound
```
Comment (miguno) [inline src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java]: Typo: remove the comma `,`

File: src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java [hunk 4]
```diff
@@ -220,10 +222,19 @@ public void shouldRemoveDuplicatesFromTheInput() throws Exception {
...
+    int numberOfSegments = 3;
+
+    // retention period must be at least window size -- for this use case, we don't need a longer retention period
+    // and thus, just use the window size as retention time
```
Comment (miguno) [inline src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java]: Typo: remove the comma `,`

File: src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java (hunk 1 above)
Comment (mjsax) [reply src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: https://github.com/apache/kafka/pull/4009

**Comment by mjsax** [comment]
Updated.

File: src/test/scala/io/confluent/examples/streams/ProbabilisticCountingScalaIntegrationTest.scala [hunk 5]
```diff
@@ -98,9 +98,9 @@ class ProbabilisticCountingScalaIntegrationTest extends AssertionsForJUnit {
     val builder: StreamsBuilder = new StreamsBuilder()
 
     val cmsStoreName = "cms-store"
-    val cmsStoreSupplier = {
+    val cmsStoreBuilder = {
       val changeloggingEnabled = true
-      val changelogConfig: util.HashMap[String, String] = {
+`      val changelogConfig: util.HashMap[String, String] = {
```
Comment (dguy) [inline src/test/scala/io/confluent/examples/streams/ProbabilisticCountingScalaIntegrationTest.scala]: is the ` accidental?

Comment (mjsax) [reply src/test/scala/io/confluent/examples/streams/ProbabilisticCountingScalaIntegrationTest.scala]: Yes... Will fix it.

**Comment by mjsax** [comment]
Updated.

File: src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java (hunk 1 above)
Comment (miguno) [reply src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: I meant: Do not leave the comment like this.  If we keep any such comment at all, then the comment should say sth like:

```java
// TODO: Code below must be changed to <short summary of the change> once KAFKA-5651 is merged.
```

Comment (mjsax) [reply src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: The comment is just for now (and for the reviewers) -- I will not merge until I can resolve this TODO :)

Comment (miguno) [reply src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: Aaaah, I see.  Then all is good.  I didn't see a "[DON'T MERGE YET]" in the PR title.

Comment (mjsax) [reply src/main/java/io/confluent/examples/streams/TopArticlesLambdaExample.java]: For docs PR, everyone merge by himself, thus not need to put into the title as I know it's not ready to get merged :) -- It's not AK where we need a committer to merge.

**Comment by mjsax** [comment]
This is replaced by #19 (to target against new `4.0.x` branch). Closing this.

**Pull request closed without merging**